### DISCUSSION TITLE
Python style guide for notebooks

### DISCUSSION THREAD

**Discussion by benwandrew** [AutoResearch/autora#416] [Core Contributions]
per the [good research code handbook](https://goodresearch.dev/tidy.html#use-the-style-guide), names for notebooks should use camel case with spaces. i tested such convention on the BSR subpackage, changing `basic-usage.ipynb` to `Basic Usage.ipynb` (the file itself and the `nav` in `mkdocs.yml`), and it seemed to build and serve successfully (see below). shall we change to this convention globally or just be consistent with what we have now?

<img width="969" alt="Screen Shot 2023-05-25 at 09 18 04" src="https://github.com/AutoResearch/autora/assets/28738595/ec9334bb-19de-4905-bdee-842afe64cfdb">

**Comment by hollandjg** [comment]
Nice. What's the URL it generates?

**Reply by benwandrew** [reply]
just locally: http://127.0.0.1:8000/Basic%20Usage/

any issue with this @hollandjg?

**Reply by musslick** [reply]
I think we should change for consistency.

**Reply by musslick** [reply]
What does it say about naming .md files?

**Reply by benwandrew** [reply]
@musslick nothing explicitly, but the [repo](https://github.com/patrickmineault/codebook) for the codebook itself follows the following convention: all caps for .md files in the root, lowercase with dash for compound names in the `docs` directory.

<img width="909" alt="Screen Shot 2023-05-25 at 10 02 30" src="https://github.com/AutoResearch/autora/assets/28738595/1102562e-342b-4451-a71e-ffa1147a94f7">

<img width="817" alt="Screen Shot 2023-05-25 at 10 03 02" src="https://github.com/AutoResearch/autora/assets/28738595/d3e6bd2a-eefd-4d51-bc49-e12388da947b">

**Reply by musslick** [reply]
Great, I will adapt that then!
